core/src/test/scala/com/salesforce/op/stages/impl/preparators/MinVarianceFilterTest.scala

```diff
@@ -0,0 +1,288 @@
+      case (i, tm) => (i, tm.value.toPickListMap, tm.value.toPickListMap,
+        tm.value.map { case (k, _) => k -> math.random }.toRealMap)
+    }
+    val (mapDataFrame, id, plMap1, plMap2, doubleMap) = TestFeatureBuilder(
+      "id", "textMap1", "textMap2", "doubleMap", mapData)
+    val features = Seq(id, plMap1, plMap2, doubleMap).transmogrify()
+    val filtered = features.filterMinVariance()
+    val output = new OpWorkflow().setResultFeatures(filtered).transform(mapDataFrame)
+    output.select(filtered.name).count() shouldBe 12
+
+    val metadata = filtered.originStage.getMetadata()
+    val summary = MinVarianceSummary.fromMetadata(metadata.getSummaryMetadata())
+    summary.dropped.size shouldBe 0
+  }
+
+  it should "produce the same statistics if the same transformation is applied twice" in {
+    val plMap = textMap.map[PickListMap](_.value.toPickListMap)
+    val features = Seq(id, plMap, plMap).transmogrify()
+    val filtered = features.filterMinVariance()
+    val output = new OpWorkflow().setResultFeatures(filtered).transform(textData)
+    output.select(filtered.name).count() shouldBe 12
+
+    val metadata = filtered.originStage.getMetadata()
+    val summary = MinVarianceSummary.fromMetadata(metadata.getSummaryMetadata())
+    summary.dropped.size shouldBe 0
+  }
+
+  private def validateEstimatorOutput
+  (
+    outputColName: String,
+    model: UnaryModel[OPVector, OPVector],
+    expectedFeaturesToDrop: Seq[String]
+  ): Unit = {
+    val metadata = model.getMetadata()
+    val summary = MinVarianceSummary.fromMetadata(metadata.getSummaryMetadata())
+    val featuresKept = OpVectorMetadata(outputColName, metadata).columns.length
+
+    summary.dropped should contain theSameElementsAs expectedFeaturesToDrop
+    featuresKept equals summary.names.toSet.diff(summary.dropped.toSet).size
+    summary.featuresStatistics.variance.min shouldBe 0.0
+  }
+
+  private def validateTransformerOutput
+  (
+    outputColName: String,
+    transformedData: DataFrame,
+    expectedFeatNames: Seq[String],
+    expectedFeaturesToDrop: Seq[String]
+  ): Unit = {
+    val expectedLength = expectedFeatNames.length - expectedFeaturesToDrop.length
+    transformedData.select(outputColName).collect().foreach { case Row(features: Vector) =>
+      features.toArray.length equals expectedLength
+    }
+  }
+
+  private def getMetadata(outputColName: String, transformedData: DataFrame): Metadata = {
+    val transformedDataSchema = transformedData.schema
+    val metadataOrig = transformedDataSchema(outputColName).metadata
+    val metadata = validateEncodeDecode(metadataOrig)
+    metadata
+  }
+
+  private def validateEncodeDecode(metadata: Metadata): Metadata = {
+    val metaParam = new MetadataParam("parent", "name", "doc")
+    val jsonEncodedMeta = metaParam.jsonEncode(metadata)
+    val jsonDecodedMeta = metaParam.jsonDecode(jsonEncodedMeta)
+    jsonDecodedMeta.hashCode() shouldEqual metadata.hashCode()
```

Comment:
@clin-projects is there no other way to compare except the hashcode?

Reply:
Changed to string comparison